### PR TITLE
Add skeleton reconcilers for the core resources.

### PR DESCRIPTION
**Pull request by mattmoor**
This also hooks them up to ./cmd/controller and drops the template controllers, which weren't doing any more than this is.

### CODE REVIEW

File: pkg/reconciler/build/build.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		logger.Errorf("invalid resource key: %s", key)
+		return nil
+	}
+
+	// Get the Build resource with this namespace/name
+	_, err = c.buildsLister.Builds(namespace).Get(name)
```
Comment (ImJasonH) [inline pkg/reconciler/build/build.go]: nit:

```go
if _, err := c.buildsLister...; errors.IsNotFound(err) {
  // not found 
} else if err != nil {
  return err
}
```

File: pkg/reconciler/buildtemplate/buildtemplate.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	_, err = c.buildTemplatesLister.BuildTemplates(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		// The BuildTemplate resource may no longer exist, in which case we stop processing.
+		logger.Errorf("buildtemplate %q in work queue no longer exists", key)
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	// TODO: Meaningful reconciliation.
```
Comment (ImJasonH) [inline pkg/reconciler/buildtemplate/buildtemplate.go]: What does this mean in this context? Is there any meaningful reconciliation to do for build templates? (same with `ClusterBuildTemplate`s)

**Comment by mattmoor** [comment]
Hmm, I marked this WIP in the hopes of creating test skeletons for each of these, but we still haven't shared the [core testing bit](https://github.com/knative/serving/blob/master/pkg/reconciler/testing/table.go) in `knative/pkg` and copying it here feels gross if we're going to let them drift.

@dprotaso I believe you still have plans to improve this further?

File: pkg/reconciler/build/build.go (hunk 1 above)
Comment (mattmoor) [reply pkg/reconciler/build/build.go]: Done, although when we start acting on the other part we'll probably rehoist it.

**Comment by ImJasonH** [comment]
/lgtm
/approve

Tests will be added in a future PR once knative/pkg gains the test helpers.

**Pull request merged**